table/get_context.cc

```diff
@@ -104,11 +108,18 @@ bool GetContext::SaveValue(const ParsedInternalKey& parsed_key,
 
     auto type = parsed_key.type;
     // Key matches. Process it
-    if ((type == kTypeValue || type == kTypeMerge) &&
+    if ((type == kTypeValue || type == kTypeMerge || type == kTypeBlobIndex) &&
         range_del_agg_ != nullptr && range_del_agg_->ShouldDelete(parsed_key)) {
       type = kTypeRangeDeletion;
     }
     switch (type) {
+      case kTypeBlobIndex:
+        if (is_blob_index_ == nullptr) {
+          // Blob value not supported. Stop.
+          state_ = kBlobIndex;
+          return false;
+        }
+      // intential fallthrough
```

Comment:
I think this could be a source of potential bugs in the future, as the code is split between the two cases. Why not move the code between lines 117 -121 to be also under case kTypeValue, and keep case kTypeBlobIndex empty (of course with the comment "intentional fall through").